runtime/doc/help.txt

```diff
@@ -185,6 +185,7 @@ Other ~
 |channel.txt|		Nvim asynchronous IO
 |dev_style.txt|		Nvim style guide
 |job_control.txt|	Spawn and control multiple processes
+|luarefvim.txt|		Lua reference manual
```

Comment:
Maybe we should split off `api.txt`, `lua.txt`, and this (and the luv) reference manual to a separate section ("APIs")?